File: jena/src/main/scala/eu/ostrzyciel/jelly/convert/jena/riot/JellySubsystemLifecycle.scala
package eu.ostrzyciel.jelly.convert.jena.riot

import org.apache.jena.sys.JenaSubsystemLifecycle

/**
 * Subsystem lifecycle for registering the Jelly language in Jena.
 * See: https://jena.apache.org/documentation/notes/system-initialization.html
 */
class JellySubsystemLifecycle extends JenaSubsystemLifecycle:
  override def start(): Unit = JellyLanguage.register()

  override def stop(): Unit = ()

  // "Application code should use level above 500."
  // Not clear whether the plugins should use > 500 or < 500...
  // Let's use 501, this should make Jelly load before the other application subsystems.
  override def level() = 501

File: jena/src/main/java/eu/neverblink/jelly/convert/jena/riot/JellySubsystemLifecycle.java
package eu.neverblink.jelly.convert.jena.riot;

import org.apache.jena.sys.JenaSubsystemLifecycle;

public final class JellySubsystemLifecycle implements JenaSubsystemLifecycle {

    @Override
    public void start() {
        JellyLanguage.register();
    }

    @Override
    public void stop() {
        // No-op
    }

    @Override
    public int level() {
        // "Application code should use level above 500."
        // Not clear whether the plugins should use > 500 or < 500...
        // Let's use 501, this should make Jelly load before the other application subsystems.
        return 501;
    }
}
